### PR TITLE
Perspective-aware drawing logic for unbounded grid

### PR DESCRIPTION
**Pull request by EarlinLutz**
In geometry\src\geometry4d, new file ViewGraphicsOps with (static) method ViewGraphicsOps.announceGridLinesInView to spew grid lines with length and spacing bounded by perspective view.

Called from frontend\src\viewcontext.ts.

### CODE REVIEW

File: core/frontend/src/ViewContext.ts [hunk 1]
```diff
@@ -325,8 +324,74 @@ export class DecorateContext extends RenderContext {
     return (undefined !== thisRay.intersectionWithPlane(dotX > dotY ? planeX : planeY, thisPt)) ? lastPt.distance(thisPt) : 0.0;
   }
 
+  public drawStandardGridNew(gridOrigin: Point3d, rMatrix: Matrix3d, spacing: XAndY, gridsPerRef: number, _isoGrid: boolean = false, _fixedRepetitions?: Point2d): void {
```
Comment (pmconne) [inline core/frontend/src/ViewContext.ts:327]: Is this intentionally `@public`? The old ``drawStandardGrid` is marked `@internal`. Can we give it a more meaningful name than "new"?

File: core/frontend/src/Viewport.ts [hunk 2]
```diff
@@ -447,6 +447,8 @@ export abstract class Viewport implements IDisposable {
   public get viewDelta(): Vector3d { return this._viewingSpace.viewDelta; }
   /** Provides conversions between world and view coordinates. */
   public get worldToViewMap(): Map4d { return this._viewingSpace.worldToViewMap; }
+  /** Provides conversions between world and view coordinates. */
```
Comment (pmconne) [inline core/frontend/src/Viewport.ts:450]: No it doesn't.

File: core/geometry/src/clipping/ConvexClipPlaneSet.ts [hunk 3]
```diff
@@ -163,7 +163,7 @@ export class ConvexClipPlaneSet implements Clipper, PolygonClipper {
    * @param interior array whose boolean value is used as both the `interior` and `invisible` bits of the plane for the succeeding segment.
```
Comment (pmconne) [inline core/geometry/src/clipping/ConvexClipPlaneSet.ts:163]: Those bits default to `false` if `undefined` is supplied for this param, correct?

File: core/frontend/src/ViewContext.ts (hunk 1 above)
Comment (bbastings) [reply core/frontend/src/ViewContext.ts:327]: This is just temporary for me to look at Paul, this code isn't finished yet and is intended to replacement drawStandardGrid.

File: core/frontend/src/Viewport.ts (hunk 2 above)
Comment (EarlinLutz) [reply core/frontend/src/Viewport.ts]: corrected.  tx

File: core/geometry/src/clipping/ConvexClipPlaneSet.ts (hunk 3 above)
Comment (EarlinLutz) [reply core/geometry/src/clipping/ConvexClipPlaneSet.ts]: correct.   comment expanded now.  tx

**Review by bbastings — APPROVED** [review]

**Review by mgooding — APPROVED** [review]

**Review by pmconne — APPROVED** [review]
Looks good. @bbastings no noticeable performance impact?

**Pull request merged**